specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+        "values": [
+          {
+            "value": "NotStarted",
+            "description": "The compilation of the custom waf policy has not started"
+          },
+          {
+            "value": "InProgress",
+            "description": "The compilation of the custom waf policy is in progress"
+          },
+          {
+            "value": "Succeeded",
+            "description": "The compilation of the custom waf policy is completed successfully and can now be referenced in the nginx config."
+          },
+          {
+            "value": "Failed",
+            "description": "The compilation of the custom waf policy failed."
+          }
+        ]
+      }
+    },
+    "NginxDeploymentDefaultWafPolicyListResponse": {
+      "type": "object",
+      "title": "Nginx Deployment Default Waf Policy List Response",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxDeploymentDefaultWafPolicyProperties"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyListResponse": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy List Response",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxDeploymentWafPolicyMetadata"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyProperties": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Properties",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "content": {
+          "format": "byte",
+          "type": "string",
+          "description": "The byte content of the Policy"
+        },
+        "filepath": {
+          "type": "string",
+          "description": "The file path where the Policy is to be saved"
+        },
+        "compilingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyCompilingStatus"
+        },
+        "applyingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyApplyingStatus"
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyMetadataProperties": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Metadata Properties",
+      "properties": {
+        "filepath": {
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "compilingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyCompilingStatus"
+        },
+        "applyingState": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyApplyingStatus"
+        }
+      }
+    },
+    "NginxDeploymentDefaultWafPolicyProperties": {
```

Comment:
>NginxDeploymentDefaultWafPolicyProperties [](http://example.com/codeflow?start=5&length=41)

you mustn't use different contracts for GET and LIST item responses

This needs to be the same as NginxDeploymentWafPolicy

Reply:
Default Waf policies are server-generated readonly resources, unlike NginxDeploymentWafPolicy's. Their properties are different hence the different contracts